File: 0443-string-compression/0443-string-compression.py
class Solution(object):
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        walker, runner = 0, 0
        while runner < len(chars):
            chars[walker] = chars[runner]
            count = 1

            while runner + 1 < len(chars) and chars[runner] == chars[runner+1]:
                runner += 1
                count += 1
            
            if count > 1:
                for c in str(count):
                    chars[walker+1] = c
                    walker += 1
                
            runner += 1
            walker += 1

        return walker
        

File: 0443-string-compression/0443-string-compression.go
import "strconv"

func compress(chars []byte) int {
    w := 0
    r := 0
    for r < len(chars) {
        c, cnt := chars[r], 0
        for r < len(chars) && c == chars[r] {
            r += 1
            cnt += 1
        }

        chars[w] = c
        w += 1
        if cnt > 1 {
            for _, d := range strconv.Itoa(cnt) {
                chars[w] = byte(d)
                w += 1
            }
        }
    }
    return w
}
